detectable/src/main/java/com/blackduck/integration/detectable/detectables/yarn/Version.java

```diff
@@ -0,0 +1,40 @@
+package com.blackduck.integration.detectable.detectables.yarn;
+
+public class Version implements Comparable<Version> {
+        int major, minor, patch;
+        
+        public Version(int major) {
+            this(major, 0, 0);
+        }
+        public Version(int major, int minor) {
+            this(major, minor, 0);
+        }
+        public Version(int major, int minor, int patch) {
+            this.major = major;
+            this.minor = minor;
+            this.patch = patch;
+        }
+        public Version(String[] parts) {
+            this(   parts.length > 0 && !parts[0].isBlank()? Integer.parseInt(parts[0]) : 0,
+                    parts.length > 1 && !parts[1].isBlank()? Integer.parseInt(parts[1]) : 0, 
+                    parts.length > 2 && !parts[2].isBlank()? Integer.parseInt(parts[2]) : 0);
+        }
+        
+        @Override
+        public String toString() {
+            return major + "." + minor + "." + patch;
+        }
+
+        @Override
+        public int compareTo(Version o) {
+            if (this.major == o.major
+                    && this.minor == o.minor) {
```

Comment:
Are we missing the patch comparison here or is this for some reason purposeful?

Reply:
Good catch! Done, thanks.
Did not affect the results because of the design to fallback to sorted latest version when not equal.